product_code/lagom/shared/client/src/main/scala/de/upb/cs/uc4/shared/client/CustomExceptionSerializer.scala

```diff
@@ -0,0 +1,126 @@
+          )
+        }
+        ByteString.fromString(Json.stringify(Json.obj(
+        "type" -> per.`type`,
+        "title" -> per.title,
+        "invalidParams" -> arr
+      )))
+
+      case message: ExceptionMessage => ByteString.fromString(Json.stringify(Json.obj(
+        "name" -> message.name,
+        "detail" -> message.detail
+      ))) //If it is not one of our custom types of Exceptions, it is one of the default exception
+
+      case _ => ByteString("")
+    }
+
+    RawExceptionMessage(errorCode, MessageProtocol(Some("application/json"), None, None), messageBytes)
+  }
+
+  override def deserialize(message: RawExceptionMessage): Throwable = {
+    val messageJson =
+      try {
+        Json.parse(message.message.iterator.asInputStream)
+      } catch {
+        case NonFatal(e) =>
+          Json.obj()
+      }
+
+    //Check if the raw json contains the fields needed for a CustomException (type, title, invalidParams). If so, use our deserializer, if not, use default
+    if (messageJson.toString().contains("type") && messageJson.toString().contains("title") && messageJson.toString().contains("invalidParams")){
+      val jsonParseResult = for {
+        eType   <- (messageJson \ "type").validate[String]
+        title <- (messageJson \ "title").validate[String]
+        invalidParams <- (messageJson \ "invalidParams").validate[Seq[SimpleError]]
+
+      } yield new DetailedError(eType, title, invalidParams)
+      val detailedError = jsonParseResult match {
+        case JsSuccess(m, _) => m
+        case JsError(_) => DetailedError("Undeserializable Exception", List())
+      }
+      fromCodeAndMessageCustom(message.errorCode, detailedError)
+    }
+    else{
+      //Default serializer for Exceptions with fields "name" and "detail"
+      val jsonParseResult = for {
+        name   <- (messageJson \ "name").validate[String]
+        detail <- (messageJson \ "detail").validate[String]
+      } yield new ExceptionMessage(name, detail)
+      val exceptionMessage = jsonParseResult match {
+        case JsSuccess(m, _) => m
+        case JsError(_)      => new ExceptionMessage("UndeserializableException", message.message.utf8String)
+      }
+      fromCodeAndMessage(message.errorCode, exceptionMessage)
+    }
+
+  }
+
+  /**
+   * Override this if you wish to deserialize your own custom Exceptions.
+   *
+   * The default implementation delegates to [[TransportException.fromCodeAndMessage()]], which will return a best match
+   * Lagom built-in exception.
+   *
+   * @param transportErrorCode The transport error code.
+   * @param exceptionMessage The exception message.
+   * @return The exception.
+   */
+  override def fromCodeAndMessage(transportErrorCode: TransportErrorCode,
+                                    exceptionMessage: ExceptionMessage
+                                  ): Throwable = {
+    TransportException.fromCodeAndMessage(transportErrorCode, exceptionMessage)
+  }
+  /**
```

Comment:
Here the docs should be updated, because this text is already the doc for the lagom exceptions. Rewriting as something like: "Used to deserialize our custom exceptions".